### PR TITLE
update to use BenchmarkTools v0.0.7

### PR DESCRIPTION
**Pull request by jrevels**
This will require https://github.com/JuliaCI/BenchmarkTools.jl/pull/43 and a subsequent v0.0.7 tag for BenchmarkTools.

I'll probably have to do some manual test runs once this PR is ready.

### CODE REVIEW

**Comment by jrevels** [comment]
This requires https://github.com/JuliaLang/METADATA.jl/pull/7863 in order for Travis to pass on v0.6

**Comment by jrevels** [comment]
I'll comment here once this has been deployed to Nanosoldier (I'm doing some cleanup work to update packages right now before running some tests, so it won't be for at least another few hours).

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,8 @@
 language: julia
 os:
   - linux
-  - osx
 julia:
   - release
-  - nightly
```
Comment (tkelman) [inline .travis.yml]: Guess the nightly cron job will cover this? Or does the Nanosoldier.jl code only run on the coordinator node with a fixed release Julia?

Comment (jrevels) [inline .travis.yml]: > Or does the Nanosoldier.jl code only run on the coordinator node with a fixed release Julia?

This. I'm never going to be running the server on a non-release version of Julia (right now it's the latest v0.5 release).

**Comment by jrevels** [comment]
Alright, the new-and-improved Nanosoldier is up and running. 

Nanosoldier passed a normal PR vs. `master` benchmark job, but I don't have time to test `daily=true` before the next daily job is triggered (45 minutes from now), so I'm just gonna let Nanosoldier undergo trial by fire since there's no downside to doing so compared to the current state of affairs. Fingers crossed.

**Pull request merged**